src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala

```diff
@@ -719,16 +721,24 @@ class StammdatenInsertService(override val sysConfig: SystemConfig) extends Even
     logger.debug(s"Create Koerbe:${lieferung.id}")
     val ret: Option[Option[Lieferung]] = stammdatenWriteRepository.getAbotypById(lieferung.abotypId) map { abotyp =>
       lieferung.lieferplanungId.map { lieferplanungId =>
+
         val abos: List[Abo] = stammdatenWriteRepository.getAktiveAbos(lieferung.abotypId, lieferung.vertriebId, lieferung.datum, lieferplanungId)
         val koerbe: List[(Option[Korb], Option[Korb])] = abos map { abo =>
-          upsertKorb(lieferung, abo, abotyp)
+          // if lieferung.vertriebId is happening the same date than any other vertrieb, this condition should be true
+          if (vertriebInDelivery(abo.vertriebId, lieferung.datum)) {
+            upsertKorb(lieferung, abo, abotyp)
+          } else { (None, None) }
         }
         recalculateNumbersLieferung(lieferung)
       }
     }
     ret.flatten.getOrElse(lieferung)
   }
 
```

Comment:
In a performance point of view it would make sence to fetch the list of dates (getVertriebByDate) just once and not for every abo

Reply:
done